tardis/adapters/sites/kubernetes.py

```diff
@@ -0,0 +1,138 @@
+from kubernetes_asyncio import client as k8s_client
+from kubernetes_asyncio.client.rest import ApiException as K8SApiException
+from ...configuration.configuration import Configuration
+from ...exceptions.tardisexceptions import TardisError
+from ...interfaces.siteadapter import SiteAdapter
+from ...interfaces.siteadapter import ResourceStatus
+from ...utilities.attributedict import AttributeDict
+from ...utilities.staticmapping import StaticMapping
+
+from functools import partial
+from datetime import datetime
+from contextlib import contextmanager
+
+
+class KubernetesAdapter(SiteAdapter):
+    def __init__(self, machine_type: str, site_name: str):
+        self._configuration = getattr(Configuration(), site_name)
+        self._machine_type = machine_type
+        self._site_name = site_name
+        key_translator = StaticMapping(
+            remote_resource_uuid="uid", drone_uuid="name", resource_status="type"
+        )
+        translator_functions = StaticMapping(
+            created=lambda date: datetime.strptime(date, "%Y-%m-%dT%H:%M:%S%z"),
+            updated=lambda date: datetime.strptime(date, "%Y-%m-%dT%H:%M:%S%z"),
+            type=lambda x, translator=StaticMapping(
+                Progressing=ResourceStatus.Running,
+                Available=ResourceStatus.Running,
+                Stopped=ResourceStatus.Stopped,
+                Deleted=ResourceStatus.Deleted,
+                ReplicaFailure=ResourceStatus.Error,
+            ): translator[x],
+        )
+        self.handle_response = partial(
+            self.handle_response,
+            key_translator=key_translator,
+            translator_functions=translator_functions,
+        )
+        self._client = None
+
+    @property
+    def client(self) -> k8s_client.AppsV1Api:
+        a_configuration = k8s_client.Configuration(
+            host=self._configuration.host,
+            api_key={"authorization": self._configuration.token},
+        )
+        a_configuration.api_key_prefix["authorization"] = "Bearer"
+        a_configuration.verify_ssl = False
+        if self._client is None:
+            self._client = k8s_client.AppsV1Api(k8s_client.ApiClient(a_configuration))
+        return self._client
+
+    async def deploy_resource(
+        self, resource_attributes: AttributeDict
+    ) -> AttributeDict:
+        spec = k8s_client.V1DeploymentSpec(
+            replicas=1,
+            selector=k8s_client.V1LabelSelector(
+                match_labels={"app": self.machine_type_configuration.label}
+            ),
+            template=k8s_client.V1PodTemplateSpec(),
+        )
+        spec.template.metadata = k8s_client.V1ObjectMeta(
+            name=self.machine_type_configuration.label,
+            labels={"app": self.machine_type_configuration.label},
+        )
+        container = k8s_client.V1Container(
+            image=self.machine_type_configuration.image,
+            args=self.machine_type_configuration.args.split(","),
```

Comment:
Since the configuration is using `yaml` there is no need to use a string and split here. You should use a `yaml` SequenceNode instead.